### PR TITLE
Replaces the bee sprites.

### PR DESCRIPTION
**Pull request by Iamgoofball**
# I'd much rather we resprite bees here for /tg/, but until someone sends me better sprites these will have to do.

Current bees have a few problems.
1. You can't tell what type of bee it is easily.
2. You can't tell what reagent it is easily.
3. Both of these can pretty much only be accomplished if you're within stinging range, which by that point it's too late and you're kind of panicking, probably trying to kill the friendly tricord bee.
4. The bee sprites are near impossible to click due to being so small, and with the constant jittering of them, it's even harder to click too.

So, I replaced them with the goon bee sprites.

dreamseeker_2016-03-25_02-00-17

These sprites address all the problems.

Now, these are **iconic goon sprites**. I don't like using them for this reason. However, we need to replace the current sprites. 
### If anyone is willing to resprite bees to fix all 4 listed problems above, I'll replace the sprites in this PR with those sprites.

This also comes with some code for bee animations.

### CODE REVIEW

**Comment by RandomMarine** [comment]
I think the existing bee sprites are good. To address color visibility/clicking issues, if a bee has reagents, their icon should have a colored aura overlay on them.

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 1]
```diff
@@ -10,6 +10,22 @@
 #define BEE_POLLINATE_PEST_CHANCE		33
 #define BEE_POLLINATE_POTENTCY_CHANCE	50
 
+// code by Haine from Goonsntation
+/proc/rand_deci(var/num1 = 1, var/num2 = 0, var/num3 = 2, var/num4 = 0)
```
Comment (Jordie0608) [inline code/modules/mob/living/simple_animal/hostile/bees.dm]: This seems kinda unnecessary as a proc; just call the code itself

**Comment by optimumtact** [comment]
If I wanted to play on a server that looked like goon, I'd play on gloom

:-1:

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 2]
```diff
@@ -10,6 +10,22 @@
...
+	var/output = text2num("[rand(num1, num2)].[rand(num3, num4)]")
+	return output
+
+
+/proc/animate_bumble(var/atom/A, var/loopnum = -1, floatspeed = 10, Y1 = 3, Y2 = -3, var/slightly_random = 1)
```
Comment (Jordie0608) [inline code/modules/mob/living/simple_animal/hostile/bees.dm]: This should be a proc of the bees; no var/x in proc arguments

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 3]
```diff
@@ -10,6 +10,22 @@
...
+	if (!istype(A))
+		return
+
+	if (slightly_random)
+		floatspeed = floatspeed * rand_deci(1, 0, 1, 4)
+	animate(A, pixel_y = Y1, time = floatspeed, loop = loopnum, easing = LINEAR_EASING)
+	animate(pixel_y = Y2, time = floatspeed, loop = loopnum, easing = LINEAR_EASING)
+	return
```
Comment (Jordie0608) [inline code/modules/mob/living/simple_animal/hostile/bees.dm]: Implicit return; remove

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 4]
```diff
@@ -10,6 +10,22 @@
...
 #define BEE_POLLINATE_POTENTCY_CHANCE	50
 
+// code by Haine from Goonsntation
+/proc/rand_deci(var/num1 = 1, var/num2 = 0, var/num3 = 2, var/num4 = 0)
+	var/output = text2num("[rand(num1, num2)].[rand(num3, num4)]")
+	return output
+
+
+/proc/animate_bumble(var/atom/A, var/loopnum = -1, floatspeed = 10, Y1 = 3, Y2 = -3, var/slightly_random = 1)
+	if (!istype(A))
```
Comment (Jordie0608) [inline code/modules/mob/living/simple_animal/hostile/bees.dm]: I'm pretty sure this is unessecary

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 5]
```diff
@@ -10,6 +10,22 @@
 #define BEE_POLLINATE_PEST_CHANCE		33
 #define BEE_POLLINATE_POTENTCY_CHANCE	50
 
+// code by Haine from Goonsntation
```
Comment (Jordie0608) [inline code/modules/mob/living/simple_animal/hostile/bees.dm]: Goonsntation

**Comment by MrPerson** [comment]
Has anyone mentioned to goofball that we can't use any code or assets from goon without explicit permission from the author?

**Comment by Jordie0608** [comment]
I believe he's aware

**Comment by Iamgoofball** [comment]
i am aware hence the big text in the OP

File: code/modules/mob/living/simple_animal/hostile/bees.dm (hunk 4 above)
Comment (optimumtact) [inline code/modules/mob/living/simple_animal/hostile/bees.dm]: unnecessary

**Comment by xxalpha** [comment]
those bees look dumb and don't fit

**Comment by Lumipharon** [comment]
Current sprite look good, they are also easy to click on because you don't have to click on the bee itself. They ALSO are easy to check what chem they have, because it literally gets added to the name.

Personally I think there should be no way to remotely check what chem a bee has (ie: some kind of hand bee scanner or someshit), as you can't really have dangerous bees without it being 11/10 obvious for 15 minutes while you build up bee numbers.

File: code/modules/mob/living/simple_animal/hostile/bees.dm (hunk 4 above)
Comment (Jordie0608) [inline code/modules/mob/living/simple_animal/hostile/bees.dm]: unnSersehri?

**Comment by Ausops** [comment]
https://file.house/bAmT.dmi
https://i.imgur.com/UWiSYfu.png

a more /tg/ appropriate queen bee sprite, also feel free to not use because I will never into ingame colouring compatible sprites

they flutter about as well

**Comment by RemieRichards** [comment]
:-1: :x: :100: FUCK. OFF.
Goof, can you ONCE in your miserable fucking life /not/ steal shit from goon?

**Comment by KorPhaeron** [comment]
I think they're cute but remie made bees and remie says no so goodbye goon sprites

**Comment by Bawhoppen** [comment]
goof I don't get why you do this crap

**Comment by duncathan** [comment]
The concerns about the current sprites are reasonable. This should be our art director's decision.

**Comment by Bawhoppen** [comment]
by art director I imagine you mean WJso don't be retarded

**Comment by Cheridan** [comment]
Any goon assets need to be in their own folder, apparently you didn't listen the first 2 times i said this

**Comment by duncathan** [comment]
That works too 👌

**Comment by Iamgoofball** [comment]
Ayy i forgot because I made this PR at 3 or 4 am

**Pull request closed without merging**